### ISSUE TITLE
ReplicationInternal.stop() never executed.

### ISSUE THREAD

**Issue by hideki** [couchbase/couchbase-lite-java-core#1158]
`ReplicationInternal.stop()` is called when StateMachine state enter `STOPPING` state.
https://github.com/couchbase/couchbase-lite-java-core/blob/master/src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java#L1300

However, ReplicationInternal.stop() method does not execute if isRunning() is false. isRunning() => RUNNING, IDLE, or OFFLINE
https://github.com/couchbase/couchbase-lite-java-core/blob/master/src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java#L1089-L1090

**Comment by pasin** [comment]
Does the current replicator stop ever work before? I thought there is a unit test to test replicator stop.

**Comment by hideki** [comment]
I believe it works. But it takes long time because it does not clear batcher or some other property.

**Comment by pasin** [comment]
I remembered this now. This is a duplicate of https://github.com/couchbase/couchbase-lite-java-core/issues/1115.

**Comment by pasin** [comment]
I'm closing #1115 as this ticket has more info and discussion.

**Comment by pasin** [comment]
I'm wondering if this is related to the multiple replicators issue.

**Comment by hideki** [comment]
I strongly believe this is related with multiple replicator issue.

If my memory is precise, We set single thread mode for RequestWorkExecutor. Single thread mode is much slower than 5 threads mode. Stopping replicator could take longer (more than 60sec) because of this bug. But we have set timeout 60 to wait stop process. If timeout exhausts, new replicator starts. When capturing thread dump, stopped replicator could be not stopped yet.
